.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
```

Comment:
May want to add the `uncrustify` check from `github.com/pocc/pre-commit-hooks` and move a copy of the formatting config to this kernel repo. https://github.com/FreeRTOS/FreeRTOS/blob/master/tools/uncrustify.cfg